OpenSim/Common/Component.cpp

```diff
@@ -737,47 +737,24 @@ const Component::StateVariable* Component::
     // Must have already called initSystem.
     OPENSIM_THROW_IF_FRMOBJ(!hasSystem(), ComponentHasNoSystem);
 
-    // Split the prefix from the varName (part of string past the last "/")
-    // In the case where no "/" is found, prefix = name.
-    std::string::size_type back = name.rfind("/");
-    std::string prefix = name.substr(0, back);
-
-    // In the case where no "/" is found, this assigns varName = name.
-    // When "/" is not found, back = UINT_MAX. Then, back + 1 = 0.
-    // Subtracting by UINT_MAX is effectively adding by 1, so the next line
-    // should work in all cases except if name.length() = UINT_MAX.
-    std::string varName = name.substr(back + 1, name.length() - back);
-
-    // first assume that the state variable named belongs to this
-    // top level component
-    std::map<std::string, StateVariableInfo>::const_iterator it;
-    it = _namedStateVariableInfo.find(varName);
-
-    if (it != _namedStateVariableInfo.end()) {
-        return it->second.stateVariable.get();
-    }
+    ComponentPath svPath(name);
 
     const StateVariable* found = nullptr;
-    const Component* comp = traversePathToComponent<Component>(prefix);
-
-    if (comp) {
-        found = comp->findStateVariable(varName);
-    }
-
-    // Path not given or could not find it along given path name
-    // Now try complete search.
-    if (!found) {
```

Comment:
You removed this backstop in the case that direct traversal could not locate the Compoent at the same time as modifying the traversal algorithm. While I agree with this change, it is difficult now to know if failures occur due to traversal bugs or because the backstop has been removed   where we were relying on it to find state variables without a complete or valid path. In the past we had no strict rule or a way to enforce it, but state names were unique in our models so we were probably relying on the find in old models/examples.

We should separate the two changes: traversal algorithm speedup (here) and then removing the find step, in which case we should rename this method (`findStateVariable`) or have a separate `getStateVariable(path)`  method.

Reply:
@aseth1 that change was actually part of https://github.com/opensim-org/opensim-core/pull/2083, which has already been merged, and in whicih `findStateVariable()` was renamed to `traverseToStateVariable()`. I've updated this PR. Sorry for the confusion. The tests will still fail for this PR, though, because socket connectees can't be found. 

Let me know if you want me to make any changes to how state variables are found (e.g., creating a `getStateVariable()`).